### DISCUSSION TITLE
Handle multifile model results

### DISCUSSION THREAD

**Discussion by jsmariegaard** [DHI/modelskill#18] [Ideas]
It could be a combination of dfsu and dfs0 or 2d and 3d results. How to chose when you add observation? Or should the combination first happen after compare? You could merge two comparison collections...

**Comment by ecomodeller** [comment]
In most cases when you have both 2d and 3d output, the item names will still be enough to identify which output it belongs to. Dfs0 is a different story and should perhaps be handled separately.

**Comment by Hendrik1987** [comment]
This might relate to my wish to also support comparison of multidimensional data, like current profiles and wave spectra. There can be occasions for wind profiles, too, when assessing 3rd party products for our forcing.

**Comment by jsmariegaard** [comment]
I am working on __add__ and __copy__ methods as a way to handle this (and other cases):

    mr1 = ModelResult('some_file.dfs0')
    o1 = PointObservation(...)
    mr1.add_observation(o1, item=0)
    cc1 = mr1.extract()

    mr2 = ModelResult('some_other_file.dfs0')
    o2 = PointObservation('other_obs.dfs0')
    mr2.add_observation(o2, item=5)
    cc2 = mr2.extract()

    cc = cc1 + cc2

**Reply by jsmariegaard** [reply]
This will with #42 be a bit simpler:

    mr1 = ModelResult('some_file.dfs0', name="my_model")
    o1 = PointObservation(...)
    c1 = mr1.extract_observation(o1)
    
    mr2 = ModelResult('some_other_file.dfs0', name="my_model") # note using the same name
    o2 = PointObservation('other_obs.dfs0')
    c2 = mr2.extract_observation(o2)
    
    cc = c1 + c2

**Comment by ecomodeller** [comment]
Outdated
